controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -325,26 +336,47 @@ func (r *K8ssandraClusterReconciler) reconcile(ctx context.Context, kc *api.K8ss
 		}
 	}
 
-	// Reconcile Stargate across all datacenters
-	stargateAuthSchemaReconciled := false
-	for i, dcTemplate := range kc.Spec.Cassandra.Datacenters {
-		actualDc := actualDcs[i]
-		dcKey := types.NamespacedName{Namespace: actualDc.Namespace, Name: actualDc.Name}
-		logger := kcLogger.WithValues("CassandraDatacenter", dcKey)
+	kcLogger.Info("All dcs reconciled")
 
-		remoteClient, err := r.ClientCache.GetRemoteClient(dcTemplate.K8sContext)
-		if err != nil {
+	if kc.HasStargates() {
+		kcLogger.Info("Reconciling Stargate auth schema")
+		dcTemplate := kc.Spec.Cassandra.Datacenters[0]
+		if remoteClient, err := r.ClientCache.GetRemoteClient(dcTemplate.K8sContext); err != nil {
 			return ctrl.Result{}, err
+		} else if err := r.reconcileStargateAuthSchema(ctx, kc, actualDcs[0], remoteClient, kcLogger); err != nil {
+			return ctrl.Result{RequeueAfter: r.ReconcilerConfig.LongDelay}, err
 		}
+	}
 
-		if err = remoteClient.Get(ctx, dcKey, actualDc); err == nil {
-			result, err := r.reconcileStargate(ctx, kc, dcTemplate, actualDc, logger, remoteClient, &stargateAuthSchemaReconciled)
-			if !result.IsZero() || err != nil {
-				return result, err
+	if reaperKeyspaces := r.getReaperKeyspaces(kc); len(reaperKeyspaces) > 0 {
```

Comment:
I don't think it makes sense to allow multiple Reaper keyspaces, does it? The keyspace should probably be a cluster setting only and not a DC-level setting. Alternatively, do a validation check to make sure it is the across all DC templates, but IMO keeping it the DC-level would be confusing and possibly error-prone.

Reply:
Indeed, I will create a `ReaperClusterTemplate` and move the keyspaces field there. I think it would also make sense to move the secrets to this new cluster template, thus making it easier to enforce the same secret for all datacenters.